### PR TITLE
fix(apply): tolerate deleted review targets

### PR DESCRIPTION
**Pull request by steipete**
## Why

Live monitoring found [run 28709649723](https://github.com/openclaw/clawsweeper/actions/runs/28709649723) failing after a successful review because target item `openclaw/openclaw#99991` was deleted before `apply-decisions` fetched its live state. GitHub correctly returned 404, but apply treated that terminal race as a workflow failure.

## What changed

- Treat a confirmed item-level 404 as the same terminal state as an already-closed item.
- Verify repository access before archiving, so a missing/inaccessible repo still fails safely.
- Preserve comment-only semantics: record the missing item without moving its report out of the active queue.
- Keep every non-404 fetch failure fatal.
- Add focused regressions for item deletion, comment-only mode, and repository-level 404.

## Proof

Exact head: `a297b52e8e619b2ba165e066b02450650ca93050`

- Focused `test/apply-live-state.test.ts`: 3/3 passed.
- Full `pnpm run check`: 617 unit and 617 repair tests passed.
- Overall coverage: 76.87% lines, 70.80% branches, 84.13% functions.
- Formatting clean.
- AutoReview first found two edge cases; both repaired. Final AutoReview: no actionable findings, confidence 0.82.
- Live smoke against deleted `openclaw/openclaw#99991`: exit 0, source report removed, durable record archived, apply report emitted `skipped_already_closed` with reason `item not found on GitHub`.
- The smoke made no GitHub write; the only follow-up call confirmed repository access.

No changelog entry: this is internal apply-run reliability with no user-facing contract change.

### CODE REVIEW

**Comment by steipete** [comment]
@clawsweeper re-review

Please review exact head `a392369917db149cf39686542b138cd30e6245ce` against the observed 404 apply failure linked in the PR body.

**Comment by steipete** [comment]
@clawsweeper re-review exact head `a297b52e8e619b2ba165e066b02450650ca93050`. The prior review's two asks are now addressed: the regression is isolated in `test/apply-live-state.test.ts`, and a real deleted-target smoke exits 0 and archives as `skipped_already_closed`. AutoReview also found and drove fixes for comment-only semantics and repository-level 404 safety; the final AutoReview is clean.

File: src/clawsweeper.ts [hunk 1]
```diff
@@ -17554,7 +17554,27 @@ async function applyDecisionsCommand(args: Args): Promise<void> {
...
+    let liveItem: ReturnType<typeof fetchItem>;
+    try {
+      liveItem = fetchItem(number);
+    } catch (error) {
+      if (!isGitHubNotFoundError(error)) throw error;
+      // Items can be deleted after review but before apply. Treat that terminal
+      // state like an already-closed item instead of failing the whole apply run.
+      markdown = replaceFrontMatterValue(markdown, "action_taken", "skipped_already_closed");
+      markdown = replaceFrontMatterValue(markdown, "apply_checked_at", new Date().toISOString());
+      archiveClosed(markdown);
```
[the remark this answers is not in the document]
Comment (steipete) [reply src/clawsweeper.ts:17581]: Fixed in `a297b52e8e`: the 404 path now mirrors comment-only semantics at `src/clawsweeper.ts:17565`, marking the check without changing `action_taken` or archiving. `test/apply-live-state.test.ts` covers the queued-record behavior.
